pkg/charm/charm.go

```diff
@@ -0,0 +1,91 @@
+// Package charm is minimilast CLI framework inspired by cobra and urfave/cli.
+package charm
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"strings"
+)
+
+var ErrNoRun = errors.New("no run method")
+
+type Constructor func(Command, *flag.FlagSet) (Command, error)
+
+type Command interface {
+	Run([]string) error
+}
+
+type Spec struct {
+	Name          string
+	Usage         string
+	Short         string
+	Long          string
+	New           Constructor
+	Hidden        bool
+	HiddenFlags   string
+	RedactedFlags string
```

Comment:
Perhaps a comment here for what Hidden vs Redacted flags mean? I was/am confused about the difference.

Reply:
Let's move charm in "as is" then improve it.  Frankly I don't remember the difference between hidden and redacted right now :-)